server/src/main/java/org/opensearch/action/search/SearchRequest.java

```diff
@@ -160,6 +161,27 @@ public SearchRequest(String[] indices, SearchSourceBuilder source) {
         this.source = source;
     }
 
+    /**
+     * Deep clone a SearchRequest
+     *
+     * @return a copy of the current SearchRequest
+     */
+    @Override
+    public SearchRequest clone() {
```

Comment:
IllegalArgumentException doesn't seem like the right fit here.  I'd recommend not catching the exception so the downstream caller can address this failure state.

```suggestion
    public SearchRequest clone() throws IOException {
```

Reply:
We cannot add `throws IOException` for this clone method since we are overriding the `clone` method for `Object.clone` and the overridden method does not throw 'java.io.IOException'. We can probably rename this method, otherwise a `RuntimeException` like `IllegalArgumentException` could be our best choice here

Reply:
Lets not override clone then, instead make a new method that can be specific about the contract being honored, such as naming this `deepCopy()...` or consider alternatives proposed from stack overflow [1] or your forum of choice

- [1] https://stackoverflow.com/a/2158486/533057